components/gitpod-db/src/typeorm/app-installation-db-impl.ts

```diff
@@ -39,37 +39,15 @@ export class TypeORMAppInstallationDBImpl implements AppInstallationDB {
         await repo.insert(obj);
     }
 
-    protected async findAndFinishInstallation(platform: AppInstallationPlatform, installationID: string): Promise<AppInstallation | undefined> {
-        // check if we find the complementary installation entries. If so, finish the installation
+    public async findInstallation(platform: AppInstallationPlatform, installationID: string): Promise<AppInstallation | undefined> {
         const repo = await this.getRepo();
-        const installationRecords = await repo.find({ where: { platform, installationID } });
-
-        // maybe we're already done and have a finished installation
-        const finishedInstallation = installationRecords.find(r => r.state == 'installed');
-        if (!!finishedInstallation) {
-            return finishedInstallation;
-        }
-
-        // maybe we need to finish an existing/ongoing installation
```

Comment:
nit: I'd love to have some reference here as it's the only piece of code that makes the table schema understandable.
Alternatively, an issue for cleaning it/the whole registration process up would be nice.

Reply:
👍🏻